src/main/java/pro/quicksense/modules/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,216 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pro.quicksense.modules.common.CommonConstant;
+import pro.quicksense.modules.entity.EmailLoginRequest;
+import pro.quicksense.modules.entity.LoginRequest;
+import pro.quicksense.modules.entity.User;
+import pro.quicksense.modules.mapper.UserMapper;
+import pro.quicksense.modules.service.UserService;
+import pro.quicksense.modules.util.CodeUtil;
+import pro.quicksense.modules.util.EmailUtil;
+import org.springframework.http.ResponseEntity;
+import pro.quicksense.modules.util.JwtInterceptor;
+
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class UserServiceImpl extends ServiceImpl<UserMapper, User> implements UserService {
+
+    private final EmailUtil emailUtil;
+    private final CodeUtil codeUtil;
+    private final JwtInterceptor jwtInterceptor;
+
+
+    @Transactional
+    public ResponseEntity<?> registerUser(User user) {
+        // Perform all validations
+        ResponseEntity<?> validationResult = validateUser(user);
+        if (validationResult != null) {
+            return validationResult;
+        }
+
+        try {
+            BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+            user.setPassword(passwordEncoder.encode(user.getPassword()));
+            user.setCreateTime(new Date());
+            user.setStatus(CommonConstant.USER_STATUS_NORMAL);
+            this.save(user);
+        } catch (DataIntegrityViolationException e) {
+            log.error("Username or email already exists", e);
+            return ResponseEntity.status(CommonConstant.INTERNAL_SERVER_ERROR_CODE).body("Username or email already exists");
+        } catch (Exception e) {
+            log.error("Error during user registration", e);
+            return ResponseEntity.status(CommonConstant.INTERNAL_SERVER_ERROR_CODE).body("An error occurred during registration");
+        }
+
+        return ResponseEntity.ok("User registration successful");
+    }
+
+    @Override
+    public ResponseEntity<?> login(LoginRequest loginRequest) {
+        LambdaQueryWrapper<User> queryWrapper = new LambdaQueryWrapper<>();
+        queryWrapper.eq(User::getUsername, loginRequest.getUsername());
+        User user = this.getOne(queryWrapper);
+
+        // Check if the user exists and is effective
+        ResponseEntity<?> checkResult = checkUserIsEffective(user);
+        if (checkResult != null) {
+            return checkResult;
+        }
+
+        // Check if the username or password is correct
+        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+        boolean matches = passwordEncoder.matches(loginRequest.getPassword(), user.getPassword());
+        if (!matches) {
+            return ResponseEntity.status(CommonConstant.BAD_REQUEST_CODE).body("Incorrect username or password");
```

Comment:
Use the predefined HTTP status code in the `package org.springframework.http`. for example: `HttpStatus.BAD_REQUEST`.